special-pages/pages/new-tab/app/favorites/components/Favorites.js

```diff
@@ -30,65 +36,224 @@ export const FavoritesMemo = memo(Favorites);
  * @param {() => void} props.add
  */
 export function Favorites({ gridRef, favorites, expansion, toggle, openContextMenu, openFavorite, add }) {
-    const platformName = usePlatformName();
     const { t } = useTypedTranslationWith(/** @type {import('../strings.json')} */ ({}));
-    const safeArea = useDropzoneSafeArea();
-
-    const ROW_CAPACITY = 6;
 
     // see: https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/
     const WIDGET_ID = useId();
     const TOGGLE_ID = useId();
 
-    const ITEM_PREFIX = useId();
-    const placeholders = calculatePlaceholders(favorites.length, ROW_CAPACITY);
     const hiddenCount = expansion === 'collapsed' ? favorites.length - ROW_CAPACITY : 0;
+    const rowHeight = ITEM_HEIGHT + ROW_GAP;
+    const canToggleExpansion = favorites.length >= ROW_CAPACITY;
 
-    // only recompute the list
-    const items = useMemo(() => {
-        return favorites
-            .map((item, index) => {
-                return (
-                    <TileMemo
-                        url={item.url}
-                        faviconSrc={item.favicon?.src}
-                        faviconMax={item.favicon?.maxAvailableSize}
-                        title={item.title}
-                        key={item.id + item.favicon?.src + item.favicon?.maxAvailableSize}
-                        id={item.id}
-                        index={index}
+    return (
+        <div class={cn(styles.root, !canToggleExpansion && styles.bottomSpace)} data-testid="FavoritesConfigured">
+            <VirtualizedGridRows
+                WIDGET_ID={WIDGET_ID}
+                favorites={favorites}
+                rowHeight={rowHeight}
+                add={add}
+                expansion={expansion}
+                openFavorite={openFavorite}
+                openContextMenu={openContextMenu}
+            />
+            {canToggleExpansion && (
+                <div
+                    className={cn({
+                        [styles.showhide]: true,
+                        [styles.showhideVisible]: canToggleExpansion,
```

Comment:
Won’t this always be true?

Reply:
no, until you have more than 6 items, it might never be able to show the extra row :)